KeeTrayTOTP/Libraries/Base32.cs

```diff
@@ -136,5 +164,28 @@ public static byte[] Decode(string base32)
 
             return outputBuffer;
         }
+
+        public static bool IsBase32(this string input)
+        {
+            return GetUnpaddedBase32(input).All(EncodingChars.Contains);
+        }
+
+        public static bool IsBase32(this string input, out string invalidCharacters)
+        {
+            invalidCharacters = null;
+            foreach (var c in GetUnpaddedBase32(input))
+            {
+                if (!EncodingChars.Contains(c))
```

Comment:
This will add ``=`` to invalidCharacters if the invalid character(s) comes after the ``=``, because ``GetUnpaddedBase32`` only trims the ``=`` at the end.

Reply:
I did notice that, but I think the way it is currently validating is not really ideal. The previous version of the method returned the numerical value of the char (hence the -1's). That was wrong already.

Some ways we could do validation:
* As we validate from left to right, we could also indicate at which point the seed becomes invalid. In this example at the first occurence of the !. 
* Keep it like this and report only the characters that are never valid. I'd like to report each invalid character once. Downside: When a seed starts with = this implementation would still say the "IsBase32", which seems wrong imo. Maybe have as special validation, that padding characters can only  appear in the end.

What are your thoughts?

Reply:
Maybe we shouldn't return the wrong characters and just state the valid characters in the GUI (a tooltip or label).

Reply:
I've added some validation that first checks if invalid padding is applied (an = appearing anywhere else then at the end). If that succeeds, it will report the invalid characters.